tests/ui/manual_flatten.stderr

```diff
@@ -0,0 +1,44 @@
+error: unnecessary `if let` since only the `Some` variant of the iterator element is used
+  --> $DIR/manual_flatten.rs:6:5
+   |
+LL | /     for n in x {
+LL | |         if let Some(n) = n {
+LL | |             println!("{}", n);
+LL | |         }
+LL | |     }
+   | |_____^ help: try: `x.into_iter().flatten()`
```

Comment:
So I forgot, that the `if let` also has to be removed. I would still use the `span_lint_and_sugg` with `MaybeIncorrect`, because it is a code suggestion, but use the span of `x` and not the complete for loop for the suggestion. This is possible with `span_lint_and_then`.

For future reference: Please add a test case where in `if let Some(x) = y` the unpacked binding name is different from the for loop binding. This will be of concern once someone tries to build a auto applicable suggestion.